rust/kernel/allocator.rs

```diff
@@ -12,12 +12,14 @@ pub struct KernelAllocator;
 
 unsafe impl GlobalAlloc for KernelAllocator {
     unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        // Safety: Returns a raw Pointer. Will return NULL in case of error.
         // `krealloc()` is used instead of `kmalloc()` because the latter is
         // an inline function and cannot be bound to as a result.
         unsafe { bindings::krealloc(ptr::null(), layout.size(), bindings::GFP_KERNEL) as *mut u8 }
     }
 
     unsafe fn dealloc(&self, ptr: *mut u8, _layout: Layout) {
+        // Safety: Freeing memory not allocated with kmalloc() can cause undefined behavior.
```

Comment:
As you noted `kfree` should only be called on memory allocated with `kmalloc`. However, as stated in the [`GlobalAlloc::dealloc` docs](https://doc.rust-lang.org/std/alloc/trait.GlobalAlloc.html#tymethod.dealloc) the caller has to guarantee that:
> * `ptr` must denote a block of memory currently allocated via this allocator,
> * `layout` must be the same layout that was used to allocate that block of memory.

So the `SAFETY` comment could look like:
```suggestion
        // SAFETY: the caller must guarantee that `ptr` and `layout` denote memory allocated with `kmalloc`.
```

Reply:
I have made the edits as suggested.